src/services/database/cascadeDelete.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-param-reassign */
+import schemas from '../../schemas';
+import relationshipFinder from '../utils/relationshipFinder';
+import patch from '../utils/patchRelationship';
+
+const promise = [];
+
+const traverseDelete = (record, table, schema) => {
+  if (!record.meta) return;
+  record.meta.archived = true;
+
+  const relationships = relationshipFinder(table, schema);
+
+  for (const link of relationships) {
+    if (link.table === table) {
+      return;
```

Comment:
If you're going to `return` immediately, don't place it above an `else if`. Short circuit it immediately: 

```
if (!('link' in table')) return;
```

That way the indentations don't get too insane, and curly braces are hard on the eyes.